src/query/functions/temporal/aggregation.go

```diff
@@ -84,78 +84,101 @@ type aggNode struct {
 }
 
 func (a *aggNode) Process(values []float64) float64 {
-	switch a.op.operatorType {
-	case AvgTemporalType:
-		var mean, count float64
-		for _, v := range values {
-			if !math.IsNaN(v) {
-				count++
-				mean += (v - mean) / count
-			}
-		}
-		return mean
-
-	case CountTemporalType:
-		var count float64
-		for _, v := range values {
-			if !math.IsNaN(v) {
-				count++
-			}
+	return aggFuncs[a.op.operatorType](values)
+}
+
+func avgOverTime(values []float64) float64 {
+	var sum, count float64
+	for _, v := range values {
+		if !math.IsNaN(v) {
+			count++
+			sum += v
 		}
-		return count
-
-	case MinTemporalType:
-		min := math.Inf(1)
-		for _, v := range values {
-			if !math.IsNaN(v) {
-				min = math.Min(min, v)
-			}
+	}
+	if count == 0 {
```

Comment:
nit: newlines

Also consider doing something like aggregation functions do where you have a `sumAndCount` function which returns `(sum, count float64)`, making a bunch of these easier